### PR TITLE
CNV-3580 expose cd-rom

### PR DESCRIPTION
**Pull request by lmandavi**
This PR is for expose cdrom in VM wizard feature:
https://issues.redhat.com/browse/CNV-3580

It is based on the design spec and follow-up clarifications by email:
https://docs.google.com/document/d/1ZtZAUg5IrUtHEFrpSDPOc9bTR1B0ASj6k3t-DL60ko4/edit#heading=h.kbcp7mxmn7ws

### CODE REVIEW

**Comment by lmandavi** [comment]
Requesting developer review.

File: modules/cnv-edit-cdrom-vm.adoc [hunk 1]
```diff
@@ -24,17 +24,22 @@ Use the following procedure to configure CD-ROMs for virtual machines.
...
+
 . In the *Edit CD-ROM* window, do the following:
-.. Select the type of CD-ROM configuration in the *Media Type* field.
-* On Windows systems, *Windows guest tools* is attached by default in
-the *Media Type* field.
-.. Complete the required information for each *Media Type*.
-.. Click *Add CD-ROM*.
-. When all CD-ROMs are added, click *Save*.
+
+* Select the type of CD-ROM configuration from the drop-down list for *Type*. CD-ROM configuration types are  *Container*, *Persistent volume clain*, *URL*, and *Windows guest tools*. *Windows guest tools* is available if the operating system is Windows.
```
Comment (gouyang) [inline modules/cnv-edit-cdrom-vm.adoc:38]: It's "Persistent Volume Claim" here.

File: modules/cnv-cdrom-wizard-fields-web.adoc [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+[id="cnv-cdrom-wizard-fields-web_{context}"]
+= CD-ROM fields
+
+|===
+|Media Type | Description
+
+|Container
+|Specify the container path. For example: kubevirt/fedora-registry-disk: latest.
+
+|Persistent volume claim
```
Comment (gouyang) [inline modules/cnv-cdrom-wizard-fields-web.adoc:15]: It's "Attach Disk" here, not "Persistent volume claim".

There is a bug for it: https://bugzilla.redhat.com/show_bug.cgi?id=1813135

File: modules/cnv-cdrom-wizard-fields-web.adoc [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+|Select the desired persistent volume claim from the drop-down list.
+
+|URL
+|Specify the URL path and size in GiB. Then, select the storage class for this URL from the drop-down list.
+
+|Windows guest tools
```
Comment (gouyang) [inline modules/cnv-cdrom-wizard-fields-web.adoc:21]: "Windows guest tools" is not a CD-ROM source on wizard, the windows-guest-tools is attached automatically if it's windows. This type should be removed.

File: modules/cnv-edit-cdrom-vm.adoc (hunk 1 above)
Comment (gouyang) [reply modules/cnv-edit-cdrom-vm.adoc:38]: The comma in `*URL*,` is not necessary.

Comment (lmandavi) [reply modules/cnv-edit-cdrom-vm.adoc:38]: Made the changes.

File: modules/cnv-cdrom-wizard-fields-web.adoc [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * cnv/cnv_virtual_machines/cnv-create-vms.adoc
+// * cnv/cnv_vm_templates/cnv-creating-vm-template.adoc
+
+[id="cnv-cdrom-wizard-fields-web_{context}"]
+= CD-ROM fields
+
+|===
+|Media Type | Description
```
Comment (jelkosz) [inline modules/cnv-cdrom-wizard-fields-web.adoc:10]: in wizard this is actually called "Media Type"

File: modules/cnv-cdrom-wizard-fields-web.adoc (hunk 2 above)
Comment (jelkosz) [reply modules/cnv-cdrom-wizard-fields-web.adoc:15]: yep indeed.

File: modules/cnv-cdrom-wizard-fields-web.adoc (hunk 3 above)
Comment (jelkosz) [reply modules/cnv-cdrom-wizard-fields-web.adoc:21]: correct

File: modules/cnv-edit-cdrom-vm.adoc (hunk 1 above)
Comment (lmandavi) [reply modules/cnv-edit-cdrom-vm.adoc:38]: Made the changes.

File: modules/cnv-cdrom-wizard-fields-web.adoc (hunk 2 above)
Comment (lmandavi) [reply modules/cnv-cdrom-wizard-fields-web.adoc:15]: Made the changes to the content to align with the notes provided in https://bugzilla.redhat.com/show_bug.cgi?id=1813135.

File: modules/cnv-cdrom-wizard-fields-web.adoc (hunk 3 above)
Comment (lmandavi) [reply modules/cnv-cdrom-wizard-fields-web.adoc:21]: Removed Windows guest tools.

File: modules/cnv-cdrom-wizard-fields-web.adoc (hunk 4 above)
Comment (lmandavi) [reply modules/cnv-cdrom-wizard-fields-web.adoc:10]: @jelkosz In the defect that Guohua noted (shown above in this PR), it was mentioned that the GUI for Wizard and the Virtual Machine page is displayed as follows:

Wizard
Source > Container / URL / Attach Disk

VM Details Page
Media Type > Container / URL / Persistent Volume

Therefore, I updated the documentation to reflect the UI notes that Guohua documented in the defect. 

Please advise should I be changing the documentation to show:

Wizard >
Media Type > Container / URL / Attach Disk

Please confirm?

**Comment by lmandavi** [comment]
@gouyang - Can you please review the changes I have made in response to your feedback and input provided by @jelkosz ? Please update me whether QE review is complete for this PR. Thanks very much!

**Comment by lmandavi** [comment]
Updated content displays in these locations:

https://cnv-3580--ocpdocs.netlify.com/openshift-enterprise/latest/cnv/cnv_virtual_machines/cnv-create-vms.html

https://cnv-3580--ocpdocs.netlify.com/openshift-enterprise/latest/cnv/cnv_virtual_machines/cnv-edit-vms.html#cnv-vm-edit-cdrom_cnv-edit-vms

**Comment by gouyang** [comment]
LGTM

File: modules/cnv-cdrom-wizard-fields-web.adoc (hunk 4 above)
Comment (jelkosz) [reply modules/cnv-cdrom-wizard-fields-web.adoc]: no, you are correct, it has to be "source" here, so it is correct as is.

**Comment by jelkosz** [comment]
LGTM

**Comment by lmandavi** [comment]
The expose CD-ROM feature applies to CNV 2.3.

File: modules/cnv-cdrom-wizard-fields-web.adoc [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+// * cnv/cnv_vm_templates/cnv-creating-vm-template.adoc
+
+[id="cnv-cdrom-wizard-fields-web_{context}"]
+= CD-ROM fields
+
+|===
+|Source | Description
+
+|Container
+|Specify the container path. For example: kubevirt/fedora-registry-disk: latest.
```
Comment (ousleyp) [inline modules/cnv-cdrom-wizard-fields-web.adoc:13]: if `kubevirt/fedora-registry-disk: latest` is something that the user types in, please put it in backticks ``

File: modules/cnv-cdrom-wizard-fields-web.adoc [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+|URL
+|Specify the URL path and size in GiB. Then, select the storage class for this URL from the drop-down list.
+
+|Attach Disk
+|Select the desired virtual machine disk.
```
Comment (ousleyp) [inline modules/cnv-cdrom-wizard-fields-web.adoc:19]: The IBM Style Guide says that "desired" should not be used. Please rephrase it to something like `Select the virtual machine disk that you want to attach.`

File: modules/cnv-edit-cdrom-vm.adoc [hunk 7]
```diff
@@ -24,17 +24,22 @@ Use the following procedure to configure CD-ROMs for virtual machines.
 .Procedure
 
 . From the *{object-gui}* tab, select your {object}.
+
 . Select the *Overview* tab.
-. Click the pencil icon to the right of the *CD-ROMs* label to open the *Edit CD-ROM* window.
-* If no CD-ROMs are available for editing, the CD label initially displays *blank*.
-* If there are CD-ROMs available, you can eject the CD-ROM by clicking *Eject CD-ROM* or remove it by clicking *-*.
+
+. To add or edit a CD-ROM configuration, click the pencil icon to the right of the *CD-ROMs* label to open the *Edit CD-ROM* window.
```
Comment (ousleyp) [inline modules/cnv-edit-cdrom-vm.adoc:30]: This sentence is a little hard to follow. I would try splitting it into two sentences.

File: modules/cnv-edit-cdrom-vm.adoc [hunk 8]
```diff
@@ -24,17 +24,22 @@ Use the following procedure to configure CD-ROMs for virtual machines.
...
+
+* If CD-ROMs are unavailable for editing, the message displays: *The virtual machine doesn't have any CD-ROMs attached*.
```
Comment (ousleyp) [inline modules/cnv-edit-cdrom-vm.adoc:32]: s/the message/the following message

Comment (ousleyp) [reply modules/cnv-edit-cdrom-vm.adoc:32]: Is there punctuation after *attached* in the UI? If so, include it before the `*`

File: modules/cnv-edit-cdrom-vm.adoc [hunk 9]
```diff
@@ -24,17 +24,22 @@ Use the following procedure to configure CD-ROMs for virtual machines.
...
+
 . In the *Edit CD-ROM* window, do the following:
-.. Select the type of CD-ROM configuration in the *Media Type* field.
-* On Windows systems, *Windows guest tools* is attached by default in
-the *Media Type* field.
-.. Complete the required information for each *Media Type*.
-.. Click *Add CD-ROM*.
-. When all CD-ROMs are added, click *Save*.
+
+* Select the type of CD-ROM configuration from the drop-down list for *Media Type*. CD-ROM configuration types are  *Container*, *URL* and *Persistent Volume Claim*.
```
Comment (ousleyp) [inline modules/cnv-edit-cdrom-vm.adoc:38]: We use the oxford comma, so please add the `,` back after `*URL*` :)

Comment (ousleyp) [reply modules/cnv-edit-cdrom-vm.adoc:38]: s/are/include

File: modules/cnv-edit-cdrom-vm.adoc [hunk 10]
```diff
@@ -24,17 +24,22 @@ Use the following procedure to configure CD-ROMs for virtual machines.
...
+
+.. Complete the required information for each *Type*.
+
+.. When all CD-ROMs are added, click *Save*.
```
Comment (ousleyp) [inline modules/cnv-edit-cdrom-vm.adoc:42]: Should this be steps a, b, and c? I think the structure of this step is confusing.

File: modules/cnv-edit-cdrom-vm.adoc (hunk 9 above)
Comment (ousleyp) [reply modules/cnv-edit-cdrom-vm.adoc:38]: s/CD-ROM configuration/The CD-ROM configuration

**Review by ousleyp — CHANGES_REQUESTED** [review]
A few suggestions and nits. Otherwise, it's looking good. Let me know if you'd like another review, and please squash your commits. Thanks!

File: modules/cnv-cdrom-wizard-fields-web.adoc (hunk 5 above)
Comment (lmandavi) [reply modules/cnv-cdrom-wizard-fields-web.adoc:13]: Added the backticks to format the entry path example.

File: modules/cnv-cdrom-wizard-fields-web.adoc (hunk 6 above)
Comment (lmandavi) [reply modules/cnv-cdrom-wizard-fields-web.adoc:19]: Made the change.

File: modules/cnv-edit-cdrom-vm.adoc (hunk 7 above)
Comment (lmandavi) [reply modules/cnv-edit-cdrom-vm.adoc:30]: Modified by creating two sentences.

File: modules/cnv-edit-cdrom-vm.adoc (hunk 8 above)
Comment (lmandavi) [reply modules/cnv-edit-cdrom-vm.adoc:32]: Added punctuation.

File: modules/cnv-edit-cdrom-vm.adoc (hunk 9 above)
Comment (lmandavi) [reply modules/cnv-edit-cdrom-vm.adoc:38]: Added the comma back.

File: modules/cnv-edit-cdrom-vm.adoc (hunk 10 above)
Comment (lmandavi) [reply modules/cnv-edit-cdrom-vm.adoc:42]: Cleaned up the steps.

**Comment by lmandavi** [comment]
@ousleyp Thanks for the peer review edits. Edits have been made, and commits have been squashed. Please review. If all looks good, this PR can be merged and CP'ed to CNV 2.3.

File: modules/cnv-cdrom-wizard-fields-web.adoc [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+|Source | Description
+
+|Container
+|Specify the container path. For example: `kubevirt/fedora-registry-disk: latest`.
+
+|URL
+|Specify the URL path and size in GiB. Then, select the storage class for this URL from the drop-down list.
+
+|Attach Disk
+|Select the desired virtual machine disk that you want to attach.
```
Comment (ousleyp) [inline modules/cnv-cdrom-wizard-fields-web.adoc:19]: @lmandavi This still says "desired" -- please remove it

File: modules/cnv-edit-cdrom-vm.adoc [hunk 12]
```diff
@@ -24,17 +24,22 @@ Use the following procedure to configure CD-ROMs for virtual machines.
 .Procedure
 
 . From the *{object-gui}* tab, select your {object}.
+
 . Select the *Overview* tab.
-. Click the pencil icon to the right of the *CD-ROMs* label to open the *Edit CD-ROM* window.
-* If no CD-ROMs are available for editing, the CD label initially displays *blank*.
-* If there are CD-ROMs available, you can eject the CD-ROM by clicking *Eject CD-ROM* or remove it by clicking *-*.
+
+. To add or edit a CD-ROM configuration, click the pencil icon to the right of the *CD-ROMs* label. The *Edit CD-ROM* window is displayed.
```
Comment (ousleyp) [inline modules/cnv-edit-cdrom-vm.adoc:30]: s/is displayed/opens

File: modules/cnv-edit-cdrom-vm.adoc [hunk 13]
```diff
@@ -24,17 +24,22 @@ Use the following procedure to configure CD-ROMs for virtual machines.
...
+
+* If CD-ROMs are unavailable for editing, the message displays: *The virtual machine doesn't have any CD-ROMs attached.*
```
Comment (ousleyp) [inline modules/cnv-edit-cdrom-vm.adoc:32]: s/the message/the following message

**Review by ousleyp — CHANGES_REQUESTED** [review]
@lmandavi There were still a few minor issues -- please refer to my latest comments and let me know when you've made the changes so that I can merge. Thanks!

File: modules/cnv-cdrom-wizard-fields-web.adoc (hunk 11 above)
Comment (lmandavi) [reply modules/cnv-cdrom-wizard-fields-web.adoc:19]: Thanks for the catch, removed.

File: modules/cnv-edit-cdrom-vm.adoc (hunk 12 above)
Comment (lmandavi) [reply modules/cnv-edit-cdrom-vm.adoc:30]: Change made.

File: modules/cnv-edit-cdrom-vm.adoc (hunk 13 above)
Comment (lmandavi) [reply modules/cnv-edit-cdrom-vm.adoc:32]: Change made.

File: modules/cnv-cdrom-wizard-fields-web.adoc (hunk 11 above)
Comment (lmandavi) [reply modules/cnv-cdrom-wizard-fields-web.adoc]: https://cnv-3580--ocpdocs.netlify.com/openshift-enterprise/latest/cnv/cnv_virtual_machines/cnv-create-vms.html#cnv-cdrom-wizard-fields-web_cnv-create-vms

File: modules/cnv-edit-cdrom-vm.adoc (hunk 12 above)
Comment (lmandavi) [reply modules/cnv-edit-cdrom-vm.adoc]: https://cnv-3580--ocpdocs.netlify.com/openshift-enterprise/latest/cnv/cnv_virtual_machines/cnv-edit-vms.html#cnv-vm-edit-cdrom_cnv-edit-vms

File: modules/cnv-edit-cdrom-vm.adoc (hunk 13 above)
Comment (lmandavi) [reply modules/cnv-edit-cdrom-vm.adoc]: https://cnv-3580--ocpdocs.netlify.com/openshift-enterprise/latest/cnv/cnv_virtual_machines/cnv-edit-vms.html#cnv-vm-edit-cdrom_cnv-edit-vms

**Comment by lmandavi** [comment]
Edits were incorporated. Commits were squashed. Please review. Thanks!

**Review by ousleyp — APPROVED** [review]
